searx/engines/google_images.py

```diff
@@ -63,26 +63,28 @@ def request(query, params):
         'https://'
         + google_info['subdomain']
         + '/search'
-        + "?"
-        + urlencode(
-            {
-                'q': query,
-                'tbm': "isch",
-                **google_info['params'],
-                'asearch': 'isch',
-                'async': '_fmt:json,p:1,ijn:' + str(params['pageno']),
-            }
-        )
+        + '?'
+        + urlencode({'q': query, 'tbm': "isch", **google_info['params'], 'asearch': 'isch'})
+        # don't urlencode this because wildly different AND bad results
+        # pagination uses Zero-based numbering
+        + f'&async=_fmt:json,p:1,ijn:{params["pageno"] - 1}'
     )
 
     if params['time_range'] in time_range_dict:
         query_url += '&' + urlencode({'tbs': 'qdr:' + time_range_dict[params['time_range']]})
     if params['safesearch']:
         query_url += '&' + urlencode({'safe': filter_mapping[params['safesearch']]})
     params['url'] = query_url
-
     params['cookies'] = google_info['cookies']
     params['headers'].update(google_info['headers'])
+    # this ua will allow getting ~50 results instead of 10. #1641
+    params['headers'].update(
+        {
+            "User-Agent": f"NSTN/3.60.474802233.release Dalvik/2.1.0 (Linux; U; Android 12; \
```

Comment:
Please note, this is a string with a lot of whitepaces in .. I recommend something like this (untested)

        params['headers']['User-Agent'] = f'NSTN/3.60.474802233.release Dalvik/2.1.0 (Linux; U; Android 12; {google_info.get('country', 'US')}) gzip'

Reply:
###### Had different variations but `make test` / `make format.python` would complain or make it even worse.
Changed it. :+1: